src/ucp/core/ucp_worker.c

```diff
@@ -2646,6 +2660,9 @@ void ucp_worker_destroy(ucp_worker_h worker)
     ucp_worker_destroy_eps(worker, &worker->all_eps, "all");
     ucp_worker_destroy_eps(worker, &worker->internal_eps, "internal");
     ucp_am_cleanup(worker);
+    ucs_callbackq_remove_if(&worker->uct->progress_q,
+                            ucp_worker_dt_invalidate_remove_filter, NULL);
```

Comment:
does it mean we destroyed a worker before all requests on it were completed?

Reply:
yes, AFAIU it's possible in case of `TERMINATE_SIGNALED`

Reply:
i think it should be considered as bug in io_demo
in UCX, we can warn if destroying non-empty callbackq

Reply:
do you think that's ok to block on waiting all requests after `TERMINATE_SIGNALED`? in common case it may take long time

Reply:
It seems like we already do it:
https://github.com/openucx/ucx/blob/c2b0874b1dac76a6d02e5142a8eb0d61a7544df5/test/apps/iodemo/io_demo.cc#L2323
so maybe there is a deeper issue..

Reply:
> It seems like we already do it

It seems like we don't
https://github.com/openucx/ucx/blob/c2b0874b1dac76a6d02e5142a8eb0d61a7544df5/test/apps/iodemo/io_demo.cc#L2002

Reply:
hmm i think we need to still wait for requests; if we close endpoints with mode FORCE, all requests should complete

Reply:
Ok, I'll open another PR with waiting all requests in io-demo, but I think this change has to be tested separately.
Also I'm going to rework this PR to complete the request in-place without scheduling to cb-queue when nothing to invalidate

Reply:
they may be completed later, but it's guaranteed the requests will complete without waiting for the remote peer

Reply:
@yosefe you are right, this PR is useless from functionality perspective but I think It may be reworked to assert  on resources cleanup. If you think that it's redundant, free to close.

Reply:
i'm pretty sure ucs_callbackq_remove_if asserts if it's not empty; if it doesn't let's add it there

Reply:
it prints `ucs_warn`, so closing this